core/Concurrency/Lock.php

```diff
@@ -77,28 +120,40 @@ public function execute($id, $callback)
         }
     }
 
+    /**
+     * Acquires a lock with the given id using the provided TTL
+     *
+     * @param string $id
+     * @param int $ttlInSeconds
+     * @return bool
+     */
     public function acquireLock($id, $ttlInSeconds = 60)
     {
-        $this->lockKey = $this->lockKeyStart . $id;
+        $this->lockKey = $this->namespace . $id;
 
         if (mb_strlen($this->lockKey) > self::MAX_KEY_LEN) {
             // Lock key might be too long for DB column, so we hash it but leave the start of the original as well
-            // to make it more readable
+            // to make it more readable and to ensure the namespaceis still containe
             $md5Len = 32;
-            $this->lockKey = mb_substr($id, 0, self::MAX_KEY_LEN - $md5Len - 1) . md5($id);
+            $this->lockKey = mb_substr($this->lockKey, 0, self::MAX_KEY_LEN - $md5Len - 1) . md5($id);
         }
 
         $lockValue = substr(Common::generateUniqId(), 0, 12);
         $locked    = $this->backend->setIfNotExists($this->lockKey, $lockValue, $ttlInSeconds);
         if ($locked) {
             $this->lockValue = $lockValue;
-            $this->lastExpireTime = Date::getNowTimestamp();
+            $this->lastAcquireTime = Date::getNowTimestamp();
         }
 
-        return $locked;
+        return !!$locked;
     }
 
-    public function isLocked()
+    /**
+     * Return if the acquired lock is being locked
```

Comment:
nitpick - 'being' indicates something in process whereas I think we want a point in time state
```suggestion
     * Return if the acquired lock is currently locked
```